judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/submission/bundle/ContestItemSubmissionResource.java

```diff
@@ -235,22 +249,63 @@ public ContestantAnswersResponse getLatestContestantAnswersInContest(
             submissions = submissions.stream().map(ItemSubmission::withoutGrading).collect(Collectors.toList());
         }
 
-        Set<String> userJids = submissions.stream().map(ItemSubmission::getUserJid).collect(Collectors.toSet());
-        Set<String> problemJids = submissions.stream().map(ItemSubmission::getProblemJid).collect(Collectors.toSet());
+        Map<String, ItemSubmission> submissionsByItemJid = submissions.stream()
+                .collect(Collectors.toMap(ItemSubmission::getItemJid, Function.identity()));
+
+        List<String> bundleProblemJidsSortedByAlias = problemStore.getProblemJids(contestJid).stream()
+                .filter(problemJid -> problemClient.getProblemInfo(problemJid).getType().equals(ProblemType.BUNDLE))
+                .collect(Collectors.toList());
+        Map<String, String> problemAliasesByProblemJid = problemStore.getProblemAliasesByJids(
+                contest.getJid(), ImmutableSet.copyOf(bundleProblemJidsSortedByAlias));
+
+        Map<String, Integer> itemNumberByItemJid = new HashMap<>();
+        Map<String, List<ItemSubmission>> submissionsByProblemJid = new HashMap<>();
+        for (String problemJid : bundleProblemJidsSortedByAlias) {
+            ProblemWorksheet worksheet = problemClient.getBundleProblemWorksheet(problemJid, language);
+            List<Item> items = worksheet.getItems().stream()
+                    .filter(item -> !item.getType().equals(ItemType.STATEMENT))
+                    .collect(Collectors.toList());
+            items.sort(Comparator.comparingInt(item -> item.getNumber().orElse(0)));
+
+            List<ItemSubmission> problemSubmissions = new ArrayList<>();
+            for (Item item : items) {
+                problemSubmissions.add(submissionsByItemJid.computeIfAbsent(
+                        item.getJid(),
+                        itemJid -> new ItemSubmission.Builder()
+                                .jid("")
```

Comment:
So many empty strings here?

Reply:
For dummy item; will be changed according to the new format (
```
    Profile getProfile();
    ContestSubmissionConfig getConfig();
    Map<String, List<String>> getItemJidsByProblemJid();
    Map<String, ItemSubmission> getSubmissionsByItemJid();
    Map<String, String> getProblemAliasesMap();
    Map<String, String> getProblemNamesMap();
```
)